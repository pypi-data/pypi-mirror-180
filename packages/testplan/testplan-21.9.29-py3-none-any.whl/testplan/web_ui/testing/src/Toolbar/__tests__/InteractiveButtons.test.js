/**
 * Unit tests for the InteractiveButtons module
 */
import React from "react";
import {shallow} from "enzyme";
import {StyleSheetTestUtils} from "aphrodite";

import {ReloadButton, ResetButton, AbortButton} from "../InteractiveButtons";

describe("ReloadButton", () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it("Renders a clickable button", () => {
    const reloadCbk = jest.fn();
    const button = shallow(
      <ReloadButton reloadCbk={reloadCbk} reloading={false} />
    );
    expect(button).toMatchSnapshot();
    button.find({title: "Reload code"}).simulate("click");
    expect(reloadCbk.mock.calls.length).toBe(1);
  });

  it("Renders a spinning icon when reload is in-progress", () => {
    const reloadCbk = jest.fn();
    const button = shallow(
      <ReloadButton reloadCbk={reloadCbk} reloading={true} />
    );
    expect(button).toMatchSnapshot();
  });

});

describe("ResetButton", () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it("Renders a clickable button", () => {
    const resetCbk = jest.fn();
    const button = shallow(
      <ResetButton resetStateCbk={resetCbk} resetting={false} />
    );
    expect(button).toMatchSnapshot();
    button.find({title: "Reset state"}).simulate("click");
    expect(resetCbk.mock.calls.length).toBe(1);
  });

  it("Renders a spinning icon when reset is in-progress", () => {
    const resetCbk = jest.fn();
    const button = shallow(
      <ResetButton resetStateCbk={resetCbk} resetting={true} />
    );
    expect(button).toMatchSnapshot();
  });

});

describe("AbortButton", () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it("Renders a clickable button", () => {
    const abortCbk = jest.fn();
    const button = shallow(
      <AbortButton abortCbk={abortCbk} aborting={false} />
    );
    expect(button).toMatchSnapshot();
    button.find({title: "Abort Testplan"}).simulate("click");
    expect(abortCbk.mock.calls.length).toBe(1);
  });

  it("Renders a spinning icon when abortion is in-progress", () => {
    const abortCbk = jest.fn();
    const button = shallow(
      <AbortButton abortCbk={abortCbk} aborting={true} />
    );
    expect(button).toMatchSnapshot();
  });

});
