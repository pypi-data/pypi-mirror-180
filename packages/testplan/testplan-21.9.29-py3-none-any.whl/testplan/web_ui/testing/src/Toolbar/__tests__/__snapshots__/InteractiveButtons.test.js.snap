// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbortButton Renders a clickable button 1`] = `
<NavItem
  key="abort-button"
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1ueb7gu"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            352,
            512,
            Array [],
            "f00d",
            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
          ],
          "iconName": "times",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-abort"
      listItem={false}
      mask={null}
      onClick={[MockFunction]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Abort Testplan"
      transform={null}
    />
  </div>
</NavItem>
`;

exports[`AbortButton Renders a spinning icon when abortion is in-progress 1`] = `
<NavItem
  key="abort-button"
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1ueb7gu-o_O-toolbarInactive_51g7gx"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            352,
            512,
            Array [],
            "f00d",
            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
          ],
          "iconName": "times",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-abort"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Aborting..."
      transform={null}
    />
  </div>
</NavItem>
`;

exports[`ReloadButton Renders a clickable button 1`] = `
<NavItem
  key="reload-button"
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1ueb7gu"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f021",
            "M440.935 12.574l3.966 82.766C399.416 41.904 331.674 8 256 8 134.813 8 33.933 94.924 12.296 209.824 10.908 217.193 16.604 224 24.103 224h49.084c5.57 0 10.377-3.842 11.676-9.259C103.407 137.408 172.931 80 256 80c60.893 0 114.512 30.856 146.104 77.801l-101.53-4.865c-6.845-.328-12.574 5.133-12.574 11.986v47.411c0 6.627 5.373 12 12 12h200.333c6.627 0 12-5.373 12-12V12c0-6.627-5.373-12-12-12h-47.411c-6.853 0-12.315 5.729-11.987 12.574zM256 432c-60.895 0-114.517-30.858-146.109-77.805l101.868 4.871c6.845.327 12.573-5.134 12.573-11.986v-47.412c0-6.627-5.373-12-12-12H12c-6.627 0-12 5.373-12 12V500c0 6.627 5.373 12 12 12h47.385c6.863 0 12.328-5.745 11.985-12.599l-4.129-82.575C112.725 470.166 180.405 504 256 504c121.187 0 222.067-86.924 243.704-201.824 1.388-7.369-4.308-14.176-11.807-14.176h-49.084c-5.57 0-10.377 3.842-11.676 9.259C408.593 374.592 339.069 432 256 432z",
          ],
          "iconName": "sync",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-reload"
      listItem={false}
      mask={null}
      onClick={[MockFunction]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Reload code"
      transform={null}
    />
  </div>
</NavItem>
`;

exports[`ReloadButton Renders a spinning icon when reload is in-progress 1`] = `
<NavItem
  key="reload-button"
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1ueb7gu-o_O-toolbarInactive_51g7gx"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f021",
            "M440.935 12.574l3.966 82.766C399.416 41.904 331.674 8 256 8 134.813 8 33.933 94.924 12.296 209.824 10.908 217.193 16.604 224 24.103 224h49.084c5.57 0 10.377-3.842 11.676-9.259C103.407 137.408 172.931 80 256 80c60.893 0 114.512 30.856 146.104 77.801l-101.53-4.865c-6.845-.328-12.574 5.133-12.574 11.986v47.411c0 6.627 5.373 12 12 12h200.333c6.627 0 12-5.373 12-12V12c0-6.627-5.373-12-12-12h-47.411c-6.853 0-12.315 5.729-11.987 12.574zM256 432c-60.895 0-114.517-30.858-146.109-77.805l101.868 4.871c6.845.327 12.573-5.134 12.573-11.986v-47.412c0-6.627-5.373-12-12-12H12c-6.627 0-12 5.373-12 12V500c0 6.627 5.373 12 12 12h47.385c6.863 0 12.328-5.745 11.985-12.599l-4.129-82.575C112.725 470.166 180.405 504 256 504c121.187 0 222.067-86.924 243.704-201.824 1.388-7.369-4.308-14.176-11.807-14.176h-49.084c-5.57 0-10.377 3.842-11.676 9.259C408.593 374.592 339.069 432 256 432z",
          ],
          "iconName": "sync",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-reload"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={true}
      symbol={false}
      title="Reloading..."
      transform={null}
    />
  </div>
</NavItem>
`;

exports[`ResetButton Renders a clickable button 1`] = `
<NavItem
  key="reset-button"
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1ueb7gu"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f049",
            "M0 436V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v151.9L235.5 71.4C256.1 54.3 288 68.6 288 96v131.9L459.5 71.4C480.1 54.3 512 68.6 512 96v320c0 27.4-31.9 41.7-52.5 24.6L288 285.3V416c0 27.4-31.9 41.7-52.5 24.6L64 285.3V436c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12z",
          ],
          "iconName": "fast-backward",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-reset"
      listItem={false}
      mask={null}
      onClick={[MockFunction]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Reset state"
      transform={null}
    />
  </div>
</NavItem>
`;

exports[`ResetButton Renders a spinning icon when reset is in-progress 1`] = `
<NavItem
  key="reset-button"
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1ueb7gu-o_O-toolbarInactive_51g7gx"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f049",
            "M0 436V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v151.9L235.5 71.4C256.1 54.3 288 68.6 288 96v131.9L459.5 71.4C480.1 54.3 512 68.6 512 96v320c0 27.4-31.9 41.7-52.5 24.6L288 285.3V416c0 27.4-31.9 41.7-52.5 24.6L64 285.3V436c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12z",
          ],
          "iconName": "fast-backward",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-reset"
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Resetting..."
      transform={null}
    />
  </div>
</NavItem>
`;
